### DISCUSSION TITLE
Mismatched Types

### DISCUSSION THREAD

**Discussion by JointKush** [tokio-rs/tokio#3538] [General] [answered]
Whenever I try to run my script I always get this error:

![hSu4Grs](https://user-images.githubusercontent.com/26471704/108583485-b85d0000-7307-11eb-836b-20e5003b5b6f.png)

I'm not sure why i'm running into. I swore this issue was fixed prior in a later version.

code:
```
use mongodb::bson::{
    self,
    doc,
    Bson
  };
  use tokio;
  
  #[tokio::main]
  async fn main() {
    
    let client = mongodb::Client::with_uri_str("mongodb://localhost:27017").await ? ;
    println!("Databases:");

    for name in client.list_database_names(None, None).await ? {
      println!("- {}", name);
    }
    Ok(())
  }
```
cargo.toml
```tokio = { version = "1.2.0", features = ["full"] }```
My version:
![tB6Q6aM](https://user-images.githubusercontent.com/26471704/108583565-446f2780-7308-11eb-8659-9449225a33c1.png)

**Comment by memoryruins** [comment, marked as answer]
```rust
async fn main() {
```

currently the return type is implicitly the [unit](https://doc.rust-lang.org/std/primitive.unit.html) type `-> ()`. It's suggesting a semicolon because adding that after `Ok(())` would cause the expression to return `()`, though not exactly what you seem to want (and the error is confusing because it's underlining the `tokio::main` macro and not the expression).

Since you are trying to return a result and use `?`, you likely meant to make the function's signature something like ` async fn main() -> Result<(), Box<dyn std::error::Error>>`

**Reply by JointKush** [reply]
I don't why I forgot all about that. Thanks that helped

**Comment by Darksonn** [comment]
When the macro gives this kind of poor error message, you can remove the macro, and the error message should improve.
